CONTRIBUTING.md

```diff
@@ -0,0 +1,412 @@
+  resource, only the code samples that meet the following requirements are
+  located in the `docs/[*component*]/samples` folders of the `knative/docs`
+  repo:
+
+  - _Actively maintained_ - The code sample has an active Knative team member
+    who has committed to regular maintenance of that content and ensures that
+    the code is updated and working for every product release.
+  - _Receives regular traffic_ - To avoid hosting and maintaining unused or
+    stale content, if code samples are not being viewed and fail to receive
+    attention or use, those samples will be moved into the
+    "[community maintained](https://github.com/knative/docs/tree/master/community/samples)"
+    set of samples.
+  - _Passes e2e testing_ - All code samples within `docs/[*component*]/samples`
+    folders must align with (and pass) the
+    [`e2e` tests](https://github.com/knative/docs/tree/master/test).
+
+  Depending on the Knative component covered by the code sample that you want to
+  contribute, your PR should add that sample in one of the following folders:
+
+  - Eventing samples:
+    [`/docs/eventing/samples`](https://github.com/knative/docs/tree/master/docs/eventing/samples)
+  - Serving samples:
+    [`/docs/serving/samples`](https://github.com/knative/docs/tree/master/docs/serving/samples)
+
+- **Community owned and maintained samples**: Code samples that have been
+  contributed by Knative community members. These samples might not receive
+  regular maintenance. It is possible that a sample is no longer current and is
+  not actively maintained by its original author. While we encourage a
+  contributor to maintain their content, we acknowledge that it's not always
+  possible for certain reasons, for example other commitments and time
+  constraints.
+
+While a sample might be out of date, it could still provide assistance and help
+you get up-and-running with certain use-cases. If you find that something is not
+right or contains outdated info, open an
+[Issue](https://github.com/knative/docs/issues/new). The sample might be fixed
+if bandwidth or available resource exists, or the sample might be taken down and
+archived into the last release branch where it worked.
+
+#### Choosing the correct branch
+
+It is likely that your docs contribution is either for new or changed features
+in the product, or for a fix or update existing content.
+
+- **New or changed features**: If you are adding or updating documentation for a
+  new or changed feature, you likely want to open your PR against the `master`
+  branch. All pre-release content for active Knative development belongs in
+  [`master`](https://github.com/knative/docs/tree/master/).
+
+- **Fixes and updates**: If you find an issue in a past release, for example a
+  typo or out-of-date content, you likely need to open multiple and subsequent
+  PRs. If not a followup PR, at least add the "`cherrypick` labels" to your
```

Comment:
I don't think most users have permissions to add labels?

Reply:
I don't recall, but we should be able to use some prow `/label` commands or the like to manage this. (And it would be really nice to have Prow be able to auto-cherrypick changes to release branches.)